reactnative/RNPlayground/app/models/PhotosScreenStore.test.ts

```diff
@@ -0,0 +1,79 @@
+import { mocked } from "ts-jest/utils"
+import { getSnapshot, applySnapshot, onSnapshot, ModelCreationType } from "mobx-state-tree"
+import { ContentModel, ErrorModel, PhotosScreenStoreModel, PhotosScreenUiState, PhotosScreenUiStateModel } from "./PhotosScreenStore"
+import { api } from "../services/api"
+
+jest.mock('../services/api')
+const mockApi = mocked(api)
+
+describe("PlaygroundStoreModel", () => {
+
+    beforeEach(() => {
+        mockApi.getPhotos.mockResolvedValue({ kind: 'ok', data: { photos: [], totalPages: 0, currentPage: 1 } })
+    })
+
+    it("creates a new instance of PlaygroundStore", () => {
+        const store = PhotosScreenStoreModel.create()
+        expect(store).toBeDefined()
+    })
+
+    it("load fetches first page", async () => {
+        const store = PhotosScreenStoreModel.create();
+
+        await store.load();
+
+        expect(mockApi.getPhotos.mock.calls[0][0]).toEqual(1)
+    })
+
+    it("shows loading followed by error when photo fetching fails", async () => {
+        const store = PhotosScreenStoreModel.create();
+        let states: PhotosScreenUiState[] = []
+        onSnapshot(store, ({ uiState }) => states.push(uiState as PhotosScreenUiState))
+
+        mockApi.getPhotos.mockResolvedValue({ kind: 'bad-data' })
+        await store.load();
+
+        expect(states).toStrictEqual([
```

Comment:
Is this the `verifySequence` counterpart in RN?

Reply:
I think it is just a simple object equality, I would assume that 2 arrays are strictly equal if all their items are the same in the same order